src/main/java/de/tum/in/www1/artemis/config/MetricsBean.java

```diff
@@ -113,13 +165,231 @@ private void registerWebsocketMetrics() {
                 .description("Number of users connected to this Artemis instance").register(meterRegistry);
     }
 
+    private void registerExerciseAndExamMetrics() {
+        int[] ranges = { 5, 15, 30, 45, 60, 120 };
+        for (int range : ranges) {
+            registerExerciseMetrics(range);
+            registerExamMetrics(range);
+        }
+    }
+
+    private void registerExerciseMetrics(int range) {
+        for (var exerciseType : ExerciseType.values()) {
+            Gauge.builder("artemis.scheduled.exercises.due.count", () -> this.getUpcomingDueExercisesCount(range, exerciseType)).strongReference(true)
+                    .tag("range", String.valueOf(range)).tag("exerciseType", exerciseType.toString()).description("Number of exercises ending within the next minutes")
+                    .register(meterRegistry);
```

Comment:
```suggestion
            final List<Tag> sharedTags = List.of(Tag.of("range", String.valueOf(range)), Tag.of("exerciseType", exerciseType.toString()));

            Gauge.builder("artemis.scheduled.exercises.due.count", () -> this.getUpcomingDueExercisesCount(range, exerciseType)).strongReference(true)
                    .tags(sharedTags).description("Number of exercises ending within the next minutes")
                    .register(meterRegistry);
```
The two `.tag()` calls can be deduplicated for all Gauge builders in the loop.

Reply:
Nice idea, thank you!